sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4292,6 +4292,14 @@ object SQLConf {
       .checkValue(_ >= 0, "The threshold of cached local relations must not be negative")
       .createWithDefault(64 * 1024 * 1024)
 
+  val QUERY_RESULT_CACHE_AUTO_REFRESH = buildConf("spark.sql.queryResultCache.autoRefresh.enabled")
+    .doc("When true, refresh the query result cache automatically if any of the tables " +
+      "referenced by the query gets updated. Note that, auto refresh won't be triggered if the " +
+      "table updates happen outside of this Spark application.")
```

Comment:
Hmm, if in same Spark application but in different SparkSession, will it be triggered?

Reply:
Yes, the key is that this Spark application (spark driver) needs to be aware of table updates, which is not possible if it happens in other systems or other Spark applications.